src/sig_stfl/sig_stfl.h

```diff
@@ -0,0 +1,578 @@
+	 *
+	 * @param[in]  mutex application defined mutex
+	 * @return OQS_SUCCESS or OQS_ERROR
+	 */
+	OQS_STATUS (*unlock_key)(void *mutex);
+
+	/**
+	 * Store Secret Key Function
+	 * Callback function used to securely store key data after a signature generation.
+	 * When populated, this pointer points to the application supplied secure storage function.
+	 * @param[in] sk_buf The serialized secret key data to secure store
+	 * @param[in] buf_len length of data to secure
+	 * @param[in] context application supplied data used to locate where this secret key
+	 *            is stored (passed in at the time the function pointer was set).
+	 *
+	 * @return OQS_SUCCESS or OQS_ERROR
+	 * Ideally written to secure device
+	 */
+	OQS_STATUS (*secure_store_scrt_key)(uint8_t *sk_buf, size_t buf_len, void *context);
+
+	/**
+	 * Free internal variant-specific data
+	 *
+	 * @param[in] sk The secret key represented as OQS_SIG_STFL_SECRET_KEY object
+	 * @return none
+	 */
+	void (*free_key)(OQS_SIG_STFL_SECRET_KEY *sk);
+
+	/*
+	 * Secure storage for private keys used in stateful signature schemes is outside the scope of the OQS library.
+	 * This is the responsibility of any adopting application. The application must supply
+	 * a function to for this purpose. A callback function and context data must be set in-order
+	 * to perform stateful signature generation.
+	 * The context var may contain, for example an HSM context, a filename or other such data that
+	 * is used to store the private key. This var is passed into the OQS lib when the application sets
+	 * the callback function use to save/update the private key.
+	*/
+	/**
+	 * Set Secret Key store callback Function
+	 *
+	 * @param[in] sk secret key pointer to be updated
+	 * @param[in] store_cb callback pointer
+	 * @param[in] context application data related to secret key data/identifier storage.
+	 *                    Provided when OQS_SIG_STFL_SECRET_KEY_SET_store_cb() is called.
+	 */
+	void (*set_scrt_key_store_cb)(OQS_SIG_STFL_SECRET_KEY *sk, secure_store_sk store_cb, void *context);
+} OQS_SIG_STFL_SECRET_KEY;
+
+/**
+ * Constructs an OQS_SIG_STFL object for a particular algorithm.
+ *
+ * Callers should always check whether the return value is `NULL`, which indicates either than an
+ * invalid algorithm name was provided, or that the requested algorithm was disabled at compile-time.
+ *
+ * @param[in] method_name Name of the desired algorithm; one of the names in `OQS_SIG_STFL_algs`.
+ * @return An OQS_SIG_STFL for the particular algorithm, or `NULL` if the algorithm has been disabled at compile-time.
+ */
+OQS_API OQS_SIG_STFL *OQS_SIG_STFL_new(const char *method_name);
+
+/**
+ * Keypair generation algorithm.
+ *
+ * Caller is responsible for allocating sufficient memory for `public_key` based
+ * on the `length_*` members in this object or the per-scheme compile-time macros
+ * `OQS_SIG_STFL_*_length_*`. The caller is also responsible for initializing
```

Comment:
Out of curiosity: Why is there so much responsibility put onto the user to "do things right"? Can't the library "help" (ensure there's less room for user error)?